src/twisted/enterprise/adbapi.py

```diff
@@ -458,16 +458,16 @@ def _close(self, conn):
 
     def _runInteraction(self, interaction, *args, **kw):
         conn = self.connectionFactory(self)
-        trans = self.transactionFactory(self, conn)
         try:
+            trans = self.transactionFactory(self, conn)
             result = interaction(trans, *args, **kw)
             trans.close()
             conn.commit()
             return result
         except:
             excType, excValue, excTraceback = sys.exc_info()
             try:
-                conn.rollback()
+                conn.rollback(excValue)
```

Comment:
`conn` is a DB-API object, whose interface is defined in PEP 249.  https://www.python.org/dev/peps/pep-0249/#id4  So we can't add an exception to the `rollback` method, even if some drivers (which ones?) might support it.